### PR TITLE
Edited the English doc

### PR DESCRIPTION
**Pull request by mibk**
Not being a native speaker myself, I can't guarantee correctness :-). Anyway, I've checked only the first half. (It's huge!)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -237,11 +237,11 @@ For the sake of completeness, lets take a look at an array which could be wrappe
...
+Schematic is a minimalist library and that's why it doesn't parse PHPDoc annotations and doesn't validate data types when accessing array data. PHPDoc annotations take place in Schematic *only for your IDE's needs*. Only validation is implemented to throw an exception when **accessing a property that misses related data in the wrapped array**. However, you can still access a property that has related data in the wrapped array even when it misses a PHPDoc annotation, but you get a warning from your IDE…
 
 ## <a name="associations"></a>Associations mapping
 
-Every associations mapping system must cover variants `many-to-one` and `one-to-many`. Schematic uses annotations (for an IDE) and a static array `Entry::$associations` for that job.
+Every association mapping system must cover variants `many-to-one` and `one-to-many`. Schematic uses annotations (for an IDE) and a static array `Entry::$associations` for that job.
```
Comment (klimesf) [inline docs/index.md]: ... (for *your* IDE) ...

Comment (mibk) [inline docs/index.md]: Done.

File: docs/index.md [hunk 2]
```diff
@@ -164,12 +164,12 @@ $template->author = $author;
 This article was written by author {{ author.name }} with ID {{ author.id }}.
 ```
 
-This quite usual solution has one big disadvantage. If you decide to rename the property `$name` to `$title` using a *rename* function in your IDE, you will most probably make a bug. Even the most modern IDEs today still cannot resolve access to properties in templates (Twig, Latte, Smarty…). As a result you will most likely have a property `$title` everywhere in your application, but an access to obsolete`$name` in your templates.
+This - quite usual - solution has one big disadvantage. If you decide to rename the property `$name` to `$title` using a *rename* function in your IDE, you will most probably make a bug. Even the most modern IDEs today still cannot resolve access to properties in templates (Twig, Latte, Smarty…). As a result, you will be still refering to the old variable name (`$name`) in your templates despite having your IDE done the renaming for you.
```
Comment (Tharos) [inline docs/index.md]: I'd prefer `to the old property name` for consistency reasons here.

**Comment by Tharos** [comment]
Thank you for your help! ❤

File: docs/index.md [hunk 3]
```diff
@@ -147,13 +147,13 @@ foreach ($tags as $tag) {
 }
 ```
 
-Collection `Entries` is immutable. Call of methods leading to a state change on it returns new instances of `Entries`.
+Collection `Entries` is immutable. Its methods return new instances of `Entries` rather that changing the state.
```
Comment (mibk) [inline docs/index.md]: It should've been `rather than` :) . Done.

File: docs/index.md (hunk 2 above)
Comment (mibk) [inline docs/index.md]: Done.

**Pull request merged**